object.go

```diff
@@ -370,14 +370,17 @@ func (obs *obs) Put(meta *ObjectMeta, r io.Reader, opts ...ObjectOpt) (*ObjectIn
 	}
 
 	// Create our own JS context to handle errors etc.
-	js, err := obs.js.nc.JetStream(PublishAsyncErrHandler(func(js JetStream, _ *Msg, err error) { setErr(err) }))
+	jetStream, err := obs.js.nc.JetStream(PublishAsyncErrHandler(func(js JetStream, _ *Msg, err error) { setErr(err) }))
 	if err != nil {
 		return nil, err
 	}
+
+	defer jetStream.(*js).cleanupReplySub()
```

Comment:
qq, why do we have to type assert here?

Reply:
`nc.JetStream()` returns an interface, not `*js` so I could either add `cleanupReplySub()` as interface method or type assert here - I decided not to clutter the interface with another method (eve unexported), but not sure which approach is better.

Reply:
ah right... makes sense